packages/katello/katello/hostname-change.rb

```diff
@@ -550,8 +550,10 @@ def run
         if success
           STDOUT.puts result
           STDOUT.puts 'Restarting puppet services'
-          run_cmd('/bin/systemctl try-restart puppet')
-          run_cmd('foreman-maintain service restart --only puppetserver')
+          puppet_services = ['puppet', 'puppetserver'].select do |service|
+            system("systemctl is-active #{service} --quiet")
+          end
+          run_cmd("/bin/systemctl try-restart #{puppet_services.join(' ')}") unless puppet_services.empty?
```

Comment:
Of course I only see this post merge. Perhaps it should have been:
```ruby
if puppet_services.any?
  STDOUT.puts 'Restarting puppet services'
  run_cmd("/bin/systemctl try-restart #{puppet_services.join(' ')}")
end
```

Reply:
Probably. Can follow up later/Monday, unless you want to?

Reply:
https://github.com/theforeman/foreman-packaging/pull/7417